<a href="https://colab.research.google.com/github/BrajanNieto/stats-prob-py/blob/main/01.%20Ex-Probability.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Probability Concepts - Exercises in Python**

---

This script includes exercises and practical implementations of key probability concepts using Python. It is designed to help reinforce theoretical knowledge through hands-on coding examples.

**Topics Covered:**  

* **Fundamental Probability Rules**: Addition and Multiplication rules, Conditional probability, and Bayes' theorem.  
* **Discrete and Continuous Distributions**: Binomial, Poisson, Normal, and others.  
* **Random Variables and Simulations**: Monte Carlo methods and more.  

**Author:**  
Bach. Brajan Esteban Nieto Espinoza  
[brajan.nieto@utec.edu.pe](mailto:brajan.nieto@utec.edu.pe)

---

In [1]:
#lib
import numpy as np

Three people enter an elevator on the first floor of a building with 10 floors. Each person is equally likely to choose any floor from 2 to 10 (independently). What is the probability that all three people want to go to consecutive floors?


In [17]:
np.random.randint(2,11,3)

array([7, 7, 6])

In [32]:
np.random.seed(100)
n_simulaciones = 10_000_000


resultados = np.random.randint(2, 11, (n_simulaciones, 3))
ordenados = np.sort(resultados, axis=1)
es_consecutivo = (ordenados[:, 1] - ordenados[:, 0] == 1) & (ordenados[:, 2] - ordenados[:, 1] == 1)
probabilidad = np.sum(es_consecutivo) / n_simulaciones

# Resultado
print(f"La probabilidad de que las 3 personas elijan pisos consecutivos es: {probabilidad:.6f}")


La probabilidad de que las 3 personas elijan pisos consecutivos es: 0.057647



### Solution  

#### 1. Sample Space:  
Each person has 9 possible choices (floors 2 to 10).  
For three people, the total number of combinations is:  
$9^3 = 729$

#### 2. Favorable Outcomes:  
To have consecutive floors, the chosen floors can be represented as \([x, x+1, x+2]\), where \(x\) is the smallest floor:  
- $x$ can take values from $2$ to $8$ (as $x+2 \leq 10$).
- For each \(x\), the three people can arrive at the floors in any order. There are \(3! = 6\) permutations for each set of consecutive floors.

Thus, the total number of favorable outcomes is:  
7 * 6 = 42

#### 3. Probability:  
The probability is the ratio of favorable outcomes to the total sample space:  
$P = \frac{\text{Favorable Outcomes}}{\text{Sample Space}} = \frac{42}{729} = \frac{14}{243} \approx 0.0576$